flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/windows.scala

```diff
@@ -222,36 +223,86 @@ class TumblingWindow(size: Expression) extends Window {
     * For batch tables, refer to a timestamp or long attribute.
     *
     * @param timeField time mode for streaming tables and time attribute for batch tables
-    * @return a tumbling window on event-time
+    * @return an on event-time window predefined
     */
-  def on(timeField: Expression): TumblingEventTimeWindow =
-    new TumblingEventTimeWindow(timeField, size)
+  def on(timeField: String): WindowPredefined =
+    on(ExpressionParser.parseExpression(timeField))
+}
+
+/**
+  * A window operating on event-time.
+  *
+  * @param timeField defines the time mode for streaming tables. For batch table it defines the
+  *                  time attribute on which is grouped.
```

Comment:
timeField, for batch table it defines the time attribute on which the data is grouped, while for stream table this should always be a constant value ('rowtime) to indicate this is a window grouped on event-time